flutter_readium/ios/flutter_readium/Sources/flutter_readium/FlutterReadiumPlugin+TTS.swift

```diff
@@ -215,3 +216,54 @@ extension FlutterReadiumPlugin : PublicationSpeechSynthesizerDelegate, AVTTSEngi
     }
   }
 }
+
+private func updateNowPlaying(locator: Locator, infoType: ControlPanelInfoType, publication: Publication?) {
+  let nowPlaying = NowPlayingInfo.shared
+  
+  let chapterNo = publication?.readingOrder.firstIndexWithHREF(locator.href)
+  nowPlaying.media?.chapterNumber = chapterNo
+  
+  if(infoType == .standard || infoType == .standardWCh || chapterNo == nil) {
+    standardNowPlayingInfo(chapterNo: chapterNo, infoType: infoType, publication: publication)
+  } else {
+    nonStandardNowPlayingInfo(chapterNo: chapterNo!, infoType: infoType, publication: publication)
+  }
+}
+
+private func standardNowPlayingInfo(chapterNo: Int?, infoType: ControlPanelInfoType, publication: Publication?) {
+  let authors = publication?.metadata.authors.map(\.name).joined(separator: ", ") ?? ""
+  var title = publication?.metadata.title ?? ""
+  
+  NowPlayingInfo.shared.media?.artist = authors
+  
+  if (infoType == .standardWCh && chapterNo != nil){
+    let currentChapter = publication?.readingOrder[chapterNo!].title
+    title += currentChapter != nil ? " - \(currentChapter!)" : ""
+    NowPlayingInfo.shared.media?.title = title
+  } else {
+    NowPlayingInfo.shared.media?.title = title
+  }
+  
+}
+
+private func nonStandardNowPlayingInfo(chapterNo: Int, infoType: ControlPanelInfoType, publication: Publication?) {
+  let currentChapter = publication?.readingOrder[chapterNo].title
+  let title = publication?.metadata.title ?? ""
+  
+  if(infoType == .chapterTitleAuthor || infoType == .chapterTitle){
+    
+    NowPlayingInfo.shared.media?.title = currentChapter ?? ""
+    
+    if(infoType == .chapterTitle){
+      NowPlayingInfo.shared.media?.artist = title
+    } else {
+      let authors = publication?.metadata.authors.map(\.name).joined(separator: ", ") ?? ""
+      let titleWithAuthors = "\(title) - \(authors)"
+      NowPlayingInfo.shared.media?.artist = titleWithAuthors
+    }
+    
+  } else {
+    NowPlayingInfo.shared.media?.artist = currentChapter ?? ""
+    NowPlayingInfo.shared.media?.title = title
+  }
+}
```

Comment:
These can be moved to the common extension at the bottom of `FlutterReadiumPlugin.swift`

Make sure to update the use from +Audiobook to also use re-use these (which take a chapter index param).